File: haskell/project-euler-solutions/test/P0001Spec.hs
module P0001Spec where

import P0001

import Test.Hspec (Spec, describe, context, it, shouldBe)

-- |Required for auto-discpvery
spec :: Spec
spec =
  describe "Project Euler Problem1 Solution" $ do

    describe "sumAllMultiplesOfBelow 3 5 10" $ do
      it "the sum of the multiples of 3 and 5 below 10 is 23" $ do
        show (sumAllMultiplesOfBelow 3 5 10) `shouldBe` "23"

    describe "sumAllMultiplesOfBelow 3 5 1000" $ do
      it "the sum of the multiples of 3 and 5 below 1000 is 233168" $ do
        show (sumAllMultiplesOfBelow 3 5 1000) `shouldBe` "233168"

File: typescript/project-euler-solutions/test/P0001.spec.ts
import { P0001 } from '../src/P0001';

describe('Helpers', () => {
  test('The sum of the multiples of 3 and 5 below 10 should be 23', () => {
    const ans = P0001.sumAllMultiplesOfBelow(3, 5, 10);
    expect(ans).toBe(23);
  });

  test('The sum of the multiples of 3 and 5 below 1000 should be 233168', () => {
    const ans = P0001.sumAllMultiplesOfBelow(3, 5, 1000);
    expect(ans).toBe(233168);
  });
});
